### PR TITLE
Linux bridge cni

### PR DESCRIPTION
**Pull request by SchSeba**
Create linux bridge cni plugin addon.
Create Makefile for cluster up/down/sync
Fix sync file

---
This change is [Reviewable](https://reviewable.io/reviews/phoracek/cluster-network-addons-operator/1)

### CODE REVIEW

**Comment by SchSeba** [comment]
@phoracek please review :)

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+
+cluster-up:
+	./cluster/up.sh
+
+cluster-down:
+	./cluster/down.sh
+
+cluster-sync:
+	./cluster/sync.sh
+
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply Makefile]: this required a binary to be installed on the development environment.
I will like to create a container (builder with everything we need and use it)

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply Makefile]: done

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (phoracek) [reply Makefile]: ok ok, won't block this PR with it :)

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+cluster-up:
+	./cluster/up.sh
+
+cluster-down:
+	./cluster/down.sh
+
+cluster-sync:
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply Makefile]: done

File: data/linux-bridge/000-ns.yaml [hunk 4]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: multus
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply data/linux-bridge/000-ns.yaml]: done

File: data/linux-bridge/001-linux-bridge.yaml [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      nodeSelector:
+        beta.kubernetes.io/arch: amd64
+      tolerations:
+        - key: node-role.kubernetes.io/master
+          operator: Exists
+          effect: NoSchedule
+      containers:
+        - name: cni-plugins
+          image: {{ .LinuxBridgeImage }}
+          imagePullPolicy: {{ .ImagePullPolicy}}
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply data/linux-bridge/001-linux-bridge.yaml]: done

File: pkg/network/linux-bridge.go [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	"path/filepath"
+
+	networkaddonsoperatorv1alpha1 "github.com/phoracek/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// validateLinuxBridge validates the combination of DisableMultiNetwork and AddtionalNetworks
+func validateLinuxBridge(conf *networkaddonsoperatorv1alpha1.NetworkAddonsConfigSpec) []error {
+	// TODO config must be there (maybe in crd def)
```
Comment (phoracek) [inline pkg/network/linux-bridge.go]: please drop this comment

File: pkg/network/linux-bridge.go [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"os"
+	"path/filepath"
+
+	networkaddonsoperatorv1alpha1 "github.com/phoracek/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// validateLinuxBridge validates the combination of DisableMultiNetwork and AddtionalNetworks
```
Comment (phoracek) [inline pkg/network/linux-bridge.go]: drop the comment

File: pkg/network/linux-bridge.go [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+func validateLinuxBridge(conf *networkaddonsoperatorv1alpha1.NetworkAddonsConfigSpec) []error {
+	// TODO config must be there (maybe in crd def)
+
+	return []error{}
+}
+
+// renderLinuxBridge generates the manifests of Linux Bridge
+func renderLinuxBridge(conf *networkaddonsoperatorv1alpha1.NetworkAddonsConfigSpec, manifestDir string) ([]*unstructured.Unstructured, error) {
+	if conf.Multus == nil {
```
Comment (phoracek) [inline pkg/network/linux-bridge.go]: we don't care about multus here

File: cluster/clean.sh [hunk 9]
```diff
@@ -2,10 +2,9 @@
 
 echo 'Cleaning up ...'
 
-./cluster/kubectl.sh delete --ignore-not-found -f deploy/cluster-network-addons-operator_00_namespace.yaml
 ./cluster/kubectl.sh delete --ignore-not-found -f deploy/cluster-network-addons-operator_01_crd.yaml
 ./cluster/kubectl.sh delete --ignore-not-found -f deploy/cluster-network-addons-operator_02_rbac.yaml
-./cluster/kubectl.sh delete --ignore-not-found -f deploy/cluster-network-addons-operator_03_daemonset.yaml
+./cluster/kubectl.sh delete --ignore-not-found -f deploy/cluster-network-addons-operator_00_namespace.yaml
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply cluster/clean.sh]: 1. we remove the all namespace we don't need to remove the deployment.
2. nothing special I think is better to remove the crd and the rbac before we remove the all namespace. (I can change it)

File: pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go [hunk 10]
```diff
@@ -7,14 +7,17 @@ import (
 // NetworkAddonsConfigSpec defines the desired state of NetworkAddonsConfig
 // +k8s:openapi-gen=true
 type NetworkAddonsConfigSpec struct {
-	Multus *MultusConfig `json:"multus,omitempty"`
+	Multus      *MultusConfig `json:"multus,omitempty"`
+	LinuxBridge *LinuxBridge  `json:"linuxBridge"`
```
[the remark this answers is not in the document]
Comment (SchSeba) [reply pkg/apis/networkaddonsoperator/v1alpha1/networkaddonsconfig_types.go]: done

File: pkg/network/linux-bridge.go (hunk 7 above)
Comment (SchSeba) [reply pkg/network/linux-bridge.go]: done

File: pkg/network/linux-bridge.go (hunk 6 above)
Comment (SchSeba) [reply pkg/network/linux-bridge.go]: done

File: pkg/network/linux-bridge.go (hunk 8 above)
Comment (SchSeba) [reply pkg/network/linux-bridge.go]: you are right my bad

File: cluster/clean.sh (hunk 9 above)
[the remark this answers is not in the document]
Comment (phoracek) [reply cluster/clean.sh]: ok ok, you can leave it as it is

**Pull request merged**